pandas/tests/test_frame.py

```diff
@@ -14751,6 +14751,16 @@ def test_query_builtin(self):
         result = df.query('sin > 5', engine=engine, parser=parser)
         tm.assert_frame_equal(expected, result)
 
+    def test_query_with_assign_statement(self):
+        df = DataFrame({'a': [1, 2, 3], 'b': ['a', 'b', 'c']})
+        a_before = df['a'].copy()
+        self.assertRaisesRegexp(
+            NotImplementedError, "'Assign' nodes are not implemented",
```

Comment:
Not a big deal (and you don't need to fix it here), but in an ideal world this should be something closer to `SyntaxError`, rather than `NotImplementedError`. `NotImplementedError` is really for abstract methods that should be overridden in a subclass or for features that genuinely have not been implemented yet. This is different: if `=` ever works in query it would be a bug.